src/app/tests/suites/TestGroupsCluster.yaml

```diff
@@ -0,0 +1,358 @@
+          values:
+              - name: "groupId"
+                value: 0x0001
+      response:
+          values:
+              - name: "status"
+                value: 0
+              - name: "groupId"
+                value: 0x0001
+              - name: "groupName"
+                value: "Group #1"
+
+    - label: "View Group 2 (not found)"
+      command: "ViewGroup"
+      arguments:
+          values:
+              - name: "groupId"
+                value: 0x1111
+      response:
+          values:
+              - name: "status"
+                value: 0x8B
+              - name: "groupId"
+                value: 0x1111
+
+    - label: "Add Group 2 (new)"
+      command: "AddGroup"
+      arguments:
+          values:
+              - name: "groupId"
+                value: 0x1111
+              - name: "groupName"
+                value: "Group #2"
+      response:
+          values:
+              - name: "status"
+                value: 0
+              - name: "groupId"
+                value: 0x1111
+
+    - label: "View Group 2 (new)"
+      command: "ViewGroup"
+      arguments:
+          values:
+              - name: "groupId"
+                value: 0x1111
+      response:
+          values:
+              - name: "status"
+                value: 0
+              - name: "groupId"
+                value: 0x1111
+              - name: "groupName"
+                value: "Group #2"
+
+    - label: "View Group 3 (not found)"
+      command: "ViewGroup"
+      arguments:
+          values:
+              - name: "groupId"
+                value: 0x7fff
+      response:
+          values:
+              - name: "status"
+                value: 0x8B
+              - name: "groupId"
+                value: 0x7fff
+
+    - label: "Add Group 3 (new)"
+      command: "AddGroup"
+      arguments:
+          values:
+              - name: "groupId"
+                value: 0x7fff
+              - name: "groupName"
+                value: "Group #3"
+      response:
+          values:
+              - name: "status"
+                value: 0
+              - name: "groupId"
+                value: 0x7fff
+
+    - label: "View Group 1 (existing)"
+      command: "ViewGroup"
+      arguments:
+          values:
+              - name: "groupId"
+                value: 0x01
+      response:
+          values:
+              - name: "status"
+                value: 0
+              - name: "groupId"
+                value: 0x01
+              - name: "groupName"
+                value: "Group #1"
+
+    - label: "View Group 2 (existing)"
+      command: "ViewGroup"
+      arguments:
+          values:
+              - name: "groupId"
+                value: 0x1111
+      response:
+          values:
+              - name: "status"
+                value: 0
+              - name: "groupId"
+                value: 0x1111
+              - name: "groupName"
+                value: "Group #2"
+
+    - label: "Get Group Membership 1"
```

Comment:
Note that this test fails: gets a failure response.  Might want to check why.....

Reply:
Encoding error. Fixed.

Reply:
What was the error, if I might ask?  I did not catch it during the review, which is making me worried.  ;)

Reply:
Oh, it was encoding capacity as a signed int, I see.